app/models/User.coffee

```diff
@@ -85,6 +86,19 @@ module.exports = class User extends CocoModel
   isSessionless: ->
     Boolean((utils.getQueryVariable('dev', false) or me.isTeacher()) and utils.getQueryVariable('course', false) and not utils.getQueryVariable('course-instance'))
 
+  getClientCreatorPermissions: ->
+    clientID = @get('clientCreator')
+    if !clientID and /@codeninjas.com$/i.test(me.get('email'))
+      clientID = "57fff652b0783842003fed00"  # hard coded for code ninjas since a lot of their users do not have clientCreator set
+    if clientID
+      api.apiClients.getByHandle(clientID)
+      .then((apiClient) => 
+        @set 'clientPermissions', apiClient.permissions
```

Comment:
Will this prevent prevent this User from saving, since clientPermissions aren't in user schema?

Reply:
In our JSON schema for user, we do not have `additionalProperties: false`, thats why it allows any extra properties and saves the user successfully.

Reply:
Ah, ok.  Will clientPermissions actually save to the database?

Is there a better place to save this temporary data than the me User model?  Seems a little confusing.

Reply:
Yeah, I can change it now that I am using me.canManageLicensesViaUI() to check for permissions